src/XliffTasks/Tasks/TransformTemplates.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using XliffTasks.Model;
+
+namespace XliffTasks.Tasks
+{
+    public sealed class TransformTemplates : XlfTask
+    {
+        [Required]
+        public ITaskItem[] Templates { get; set; }
+
+        [Required]
+        public ITaskItem[] UnstructuredResources { get; set; }
+
+        [Required]
+        public string[] Languages { get; set; }
+
+        [Required]
+        public string TranslatedOutputDirectory { get; set; }
+
+        [Output]
+        public ITaskItem[] LocalizedTemplates { get; set; }
+
+        [Output]
+        public ITaskItem[] NotLocalizedTemplates { get; set; }
+
+        protected override void ExecuteCore()
+        {
+            var localizedTemplates = new List<ITaskItem>();
+            var notLocalizedTemplates = new List<ITaskItem>();
+            var resourceMap = UnstructuredResources.ToDictionary(item => item.GetMetadata("FullPath"));
+            foreach (var template in Templates)
+            {
+                var translatableResourceString = template.GetMetadata("TranslatableResources");
+                if (string.IsNullOrEmpty(translatableResourceString))
+                {
+                    // if no localization is to be done, simply pass the template item through
+                    localizedTemplates.Add(template);
```

Comment:
I'm a bit confused by the naming here. Things that did *not* require any localization are designated as 'localized'.